plugins/aep-iam-consent/index.ts

```diff
@@ -0,0 +1,65 @@
+/*
+  Copyright 2023 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+import {
+  Configuration,
+  SharedStateVersions
+} from '@adobe/griffon-toolkit-aep-mobile';
+import { Event } from '@adobe/griffon-toolkit-common';
+import { ValidationPluginResult } from '../../types/validationPlugin';
+
+(function (events: Event[]): ValidationPluginResult {
+  const { toolkit: kit } = window.griffon;
+  const { configuration } = kit['aep-mobile'];
+
+  const configurations: Configuration[] = kit.match(
+    configuration.matcher,
+    events
+  );
+
+  const isConsentActive = configurations.some(
+    (event) => event.payload.metadata?.['state.data']?.['consent.default']
+  );
+
+  if (!isConsentActive) {
+    return {
+      message:
+        'You have not registered a Consent configuration in launch. Consent is not enabled',
+      events: [],
+      result: 'unknown'
+    };
+  }
+
+  const { sharedStateVersions: versions } = kit['aep-mobile'];
+  const versionEvents = kit.match(
+    versions.matcher,
+    events
+  ) as SharedStateVersions[];
+
+  const isConsentInstalled = versionEvents.some(
+    versions.getExtensionsKey('"com.adobe.edge.consent"')
+  );
+
+  if (!isConsentInstalled) {
+    return {
+      message:
+        'Consent has been configured inside launch, but the extension has not been registered in your application',
```

Comment:
Data Collection instead of launch

Reply:
Could also add https://github.com/AdobeDocs/aep-mobile-sdkdocs/blob/main/src/pages/documentation/getting-started/tabs/get-the-sdk.md as a `link`

Reply:
@dgrossen updated!